application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml

```diff
@@ -59,13 +60,24 @@
       #set($list = $obj.getProperty('flash').getValue())
       #if($list == $util.getNull())
         #set($discard = $obj.set('flash', [$xcontext.user]))
+        #set($discard = $obj.set('seenDate', $dateNow))
         #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #elseif($list.class.name == 'com.xpn.xwiki.objects.ListProperty$NotifyList' &amp;&amp; !$list.contains($xcontext.user))
         #set($discard = $list.add($xcontext.user))
         #set($discard = $obj.set('flash', $list))
+        #set($discard = $obj.set('seenDate', $dateNow))
         #set($discard = $msgdoc.saveAsAuthor("$xcontext.user has seen the message"))
       #end
     #end
+    ## Check if the message should be mark as seen today
+    #set($msgObj = $msgdoc.getObject('Flash.FlashValidationClass'))
+    #set($msgDateSeen = $msgObj.getProperty('seenDate').getValue())
+    #set($dateToday = $datetool.format('yyyy-M-d',$dateNow))
+    #set($seenDate = $datetool.format('yyyy-M-d',$msgDateSeen))
```

Comment:
Same as above about the date format. You can even check the default date format of the wiki. See: https://github.com/xwiki/xwiki-platform/commit/495e81f881e7e7d4886ba075974f607207500313#diff-b63753a40cda0f6a848b4d11bee1b84eR219.